### ISSUE TITLE
mbedtls_cert_write : "Destination buffer is too small"

### ISSUE THREAD

**Issue by jleroy** [ARMmbed/mbedtls#315]
If I use `mbedtls_cert_write`command with the `issuer_crt` parameter, I get the following error:

```
  . Seeding the random number generator... ok
  . Reading serial number... ok
  . Loading the issuer certificate ... failed
  !  mbedtls_x509_dn_gets returned -0x2980 - X509 - Destination buffer is too small 
```

Here's the full command used (anonymised): 

```
mbedtls_cert_write issuer_crt=SHA-256_Root_CA-V1.crt issuer_key=SHA-256_Root_CA-V1.key subject_key=SHA-256_Internal_CA-V1.key subject_name='C=FR,O=XXXX,OU=Certmaster,ST=XXXXX,L=XXXXX,postalAddress=XXXXX,postalCode=XXXXX,R=XXXXX,CN=SHA-256 Internal CA V1' max_pathlen=0 key_usage=key_cert_sign,crl_sign not_before=20151010010500 not_after=20201010010500 serial=XXXXX output_file=SHA-256_Internal_CA-V1.crt
```

**Comment by jleroy** [comment]
I use mbedtls 2.1.2, BTW.

**Comment by jleroy** [comment]
I have the same problem when using a CSR file: 

```
  . Loading the certificate request ... failed
  !  mbedtls_x509_dn_gets returned -0x2980 - X509 - Destination buffer is too small
```

**Comment by mpg** [comment]
Thanks for your report!

Can you try changing the definition of `issuer_name` to something larger on [line 192 of cert_write.c](https://github.com/ARMmbed/mbedtls/blob/d97f899f99e15fa058ade6d6d8ecf2bbcce74aee/programs/x509/cert_write.c#L192) and see if it fixes the problem?

**Comment by mpg** [comment]
Oh, I wrote my previous message before realising you wrote a PR for that. Thanks!

**Comment by rainierwolfcastle** [comment]
http://jira-dev0.cambridge.arm.com/browse/IOTSSL-466

**Comment by rainierwolfcastle** [comment]
ARM Internal Ref: IOTSSL-509

**Comment by jleroy** [comment]
@rainierwolfcastle FYI this bug is fixed by PR #317.
